### PR TITLE
Use `swift_add_tool()` for unit test suite

### PR DESCRIPTION
**Pull request by peddie**
This enables additional warnings.

### CODE REVIEW

File: include/albatross/src/core/model.hpp [hunk 1]
```diff
@@ -59,8 +59,8 @@ template <typename ModelType> class ModelBase : public ParameterHandlingMixin {
                 !has_possible_fit<ModelType, FeatureType>::value &&
                     !has_valid_fit<ModelType, FeatureType>::value,
                 int>::type = 0>
-  void _fit(const std::vector<FeatureType> &features,
```
Comment (akleeman) [inline include/albatross/src/core/model.hpp:62]: the other option is (I think) leaving out the variable name,
```
void _fit(const std::vector<FeatureType> &,
```

File: include/albatross/src/details/typecast.hpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  ALBATROSS_ASSERT(input >= 0);
+  return static_cast<std::size_t>(input);
+}
+
+constexpr Eigen::Index to_index(std::size_t input) {
+  ALBATROSS_ASSERT(input <= to_size(std::numeric_limits<Eigen::Index>::max()));
+  return static_cast<Eigen::Index>(input);
+}
+
+constexpr std::size_t kMaxIntegerPrecisionDoubleSize = 1ULL << 53;
```
Comment (akleeman) [inline include/albatross/src/details/typecast.hpp:30]: magic number: maybe a comment about why 53?

File: include/albatross/src/details/typecast.hpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+constexpr double to_double(std::size_t input) {
+  ALBATROSS_ASSERT(input <= kMaxIntegerPrecisionDoubleSize);
+  return static_cast<double>(input);
+}
+
+// We have to be sure to handle the case of 32-bit `Eigen::Index`
+constexpr Eigen::Index kMaxIntegerPrecisionDoubleIndex =
+    static_cast<Eigen::Index>(std::min(
+        static_cast<long long>(std::numeric_limits<Eigen::Index>::max()),
+        1LL << 53));
```
Comment (akleeman) [inline include/albatross/src/details/typecast.hpp:41]: same

File: tests/mock_model.h [hunk 4]
```diff
@@ -102,12 +101,12 @@ class MockModel : public ModelBase<MockModel> {
...
 
 static inline RegressionDataset<MockFeature>
-mock_training_data(const int n = 10) {
+mock_training_data(const Eigen::Index n = 10) {
   std::vector<MockFeature> features;
   Eigen::VectorXd targets(n);
   for (int i = 0; i < n; i++) {
     features.push_back(MockFeature(i));
-    targets[i] = static_cast<double>(i + n);
+    targets[static_cast<Eigen::Index>(i)] = cast::to_double(i + n);
```
Comment (akleeman) [inline tests/mock_model.h:109]: `to_index`?

**Review by akleeman — APPROVED** [review]

File: include/albatross/src/core/model.hpp (hunk 1 above)
Comment (peddie) [reply include/albatross/src/core/model.hpp:62]: Yes -- `clang-tidy` complains about this, however.

File: include/albatross/src/details/typecast.hpp (hunk 2 above)
Comment (peddie) [reply include/albatross/src/details/typecast.hpp:30]: Good call, will do

File: tests/mock_model.h (hunk 4 above)
Comment (peddie) [reply tests/mock_model.h:109]: Not in this case -- it's not being cast from an STL index (`std::size_t`) but from a mock feature of type `int`.  I can add some more checks for other cases like this, but I tried to focus here mainly on safely handling index conversions.

File: include/albatross/src/core/model.hpp (hunk 1 above)
Comment (akleeman) [reply include/albatross/src/core/model.hpp:62]: Ah weird, anyway this works!

File: tests/mock_model.h (hunk 4 above)
Comment (akleeman) [reply tests/mock_model.h:109]: Ah, I see.  Yeah all good, just thought it was an overlooked case.

**Pull request merged**